signer/src/bitcoin/utxo.rs

```diff
@@ -162,18 +165,39 @@ impl SbtcRequests {
             })
             .map(RequestRef::Withdrawal);
 
-        // Now we filter deposit requests where the user's max fee could be
-        // less than the fee we may charge. This is simpler because deposit
-        // UTXOs have a known fixed size. Also, the "actual" max fee must
-        // always be less than or equal to the deposit amount, so take the
-        // minimum of the max fee and the amount.
+        // Filter deposit requests based on two constraints:
+        // 1. The user's max fee must be >= our minimum required fee for deposits
+        //     (based on fixed deposit tx size)
+        // 2. The deposit amount must be less than the per-deposit limit
+        // 3. The total amount being minted must stay under the maximum allowed mintable amount
         let minimum_deposit_fee = self.compute_minimum_fee(SOLO_DEPOSIT_TX_VSIZE);
-        let deposits = self
-            .deposits
-            .iter()
-            .filter(|req| req.max_fee.min(req.amount) >= minimum_deposit_fee)
-            .map(RequestRef::Deposit);
 
```

Comment:
I might embed this logic into the `SbtcLimits` impl instead since we likely want the same kind of logic everywhere.